yt_dlp/extractor/neteasemusic.py

```diff
@@ -110,62 +76,59 @@ def extract_formats(self, info):
             details = info.get(song_format)
             if not details:
                 continue
-
             bitrate = int_or_none(details.get('bitrate')) or 999000
-            data = self._call_player_api(song_id, bitrate)
-            for song in try_get(data, lambda x: x['data'], list) or []:
-                song_url = try_get(song, lambda x: x['url'])
+            for song in traverse_obj(self._call_player_api(song_id, bitrate), ('data', ...)):
+                song_url = traverse_obj(song, ('url', {url_or_none}))
                 if not song_url:
                     continue
                 if self._is_valid_url(song_url, info['id'], 'song'):
                     formats.append({
                         'url': song_url,
-                        'ext': details.get('extension'),
-                        'abr': float_or_none(song.get('br'), scale=1000),
                         'format_id': song_format,
-                        'filesize': int_or_none(song.get('size')),
-                        'asr': int_or_none(details.get('sr')),
+                        'asr': traverse_obj(details, ('sr', {int_or_none})),
+                        **traverse_obj(song, {
+                            'ext': ('type', {str}),
+                            'abr': ('br', {lambda i: int_or_none(i, scale=1000)}),
+                            'filesize': ('size', {int_or_none}),
+                        }),
                     })
                 elif err == 0:
                     err = try_get(song, lambda x: x['code'], int)
 
         if not formats:
-            msg = 'No media links found'
             if err != 0 and (err < 200 or err >= 400):
-                raise ExtractorError(
-                    '%s (site code %d)' % (msg, err, ), expected=True)
+                raise ExtractorError(f'No media links found (site code {err})', expected=True)
             else:
-                self.raise_geo_restricted(
-                    msg + ': probably this video is not available from your location due to geo restriction.',
-                    countries=['CN'])
-
+                self.raise_geo_restricted('No media links found: probably due to geo restriction.')
```

Comment:
Right, I removed it because xff was not working. It should be safe to assume region is always `CN` if geo restricted.
Adding the countries back: https://github.com/yt-dlp/yt-dlp/pull/8181/commits/e09a1160f4fafb6c9f391de55b7445e019a3c813